src/config/mongoose_config_parser_toml.erl

```diff
@@ -301,14 +338,20 @@ cyrsasl_external(M) -> {mod, b2a(M)}.
 -spec partition_auth_opts([{atom(), any()}], ejabberd:server()) -> [config()].
 partition_auth_opts(AuthOpts, Host) ->
     {InnerOpts, OuterOpts} = lists:partition(fun({K, _}) -> is_inner_auth_opt(K) end, AuthOpts),
-    [#local_config{key = {auth_opts, Host}, value = InnerOpts} |
-     [#local_config{key = {K, Host}, value = V} || {K, V} <- OuterOpts]].
+    case InnerOpts of
+        [] ->
+            [#local_config{key = {K, Host}, value = V} || {K, V} <- OuterOpts];
+        _ ->
+            [#local_config{key = {auth_opts, Host}, value = InnerOpts} |
+                [#local_config{key = {K, Host}, value = V} || {K, V} <- OuterOpts]]
+            end.
```

Comment:
I would **not** do this change now.

To give you the motivation, let me describe how the current version works:

- If there is an `auth` section in `host_config`, it overwrites all the globally set _inner_ options, changing them to defaults unless they are specified for the host.
- The advantage is that it is quite intuitive and in line with the `cfg` logic: a section specified for the host overwrites the default section.
- The disadvantage is that only the inner option are overwritten with defaults.

The version proposed in this PR would work like this:

- If there is any of the _inner_ options in the `auth` section for a host, it overwrites all the _inner_ options, changing them to defaults unless they are specified for the host.
- The disadvantage is that it is counter-intuitive as despite having an `auth` section defined, it would overwrite the global settings only if a specific option is present.

As I do not see the proposed changes as a solution and the big tests also depend on the current state, I would not do this change now.

However, I think it needs to be resolved in such a way that the distinction between _inner_ and _outer_ options disappears completely and we need to do this when we drop the support for the `cfg` format.

I hope this clarifies my view.